springfox-core/src/main/java/springfox/documentation/schema/Model.java

```diff
@@ -96,8 +107,42 @@ public String getDiscriminator() {
   public ResolvedType getType() {
     return type;
   }
+  
+  public Boolean isMap() {
+      return isMapType;
+    }
 
   public String getExample() {
     return example;
   }
+  
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(id, name, type, qualifiedType, isMapType, properties, description, baseModel, discriminator, subTypes, example);
+  }
+  
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    Model that = (Model) o;
+
+    return Objects.equal(id, that.id) &&
+        Objects.equal(name, that.name) &&
+        Objects.equal(type, that.type) &&
+        Objects.equal(isMapType, that.isMapType) &&
+        Objects.equal(qualifiedType, that.qualifiedType) &&
+        Objects.equal(properties, that.properties) &&
+        Objects.equal(description, that.description) &&
+        Objects.equal(baseModel, that.baseModel) &&
+        Objects.equal(discriminator, that.discriminator) &&
+        Objects.equal(subTypes, that.subTypes) &&
+        Objects.equal(example, that.example);
+  }
```

Comment:
This change seems a little risky too me. I'd rather extract this to an `Equivalence` [like this](https://github.com/springfox/springfox/blob/master/springfox-spring-web/src/main/java/springfox/documentation/spring/web/OperationCachingEquivalence.java).